Library/Homebrew/dev-cmd/bottle.rb

```diff
@@ -441,9 +442,12 @@ def bottle_formula(f, args:)
           mv tar_path, relocatable_tar_path
           # Use gzip, faster to compress than bzip2, faster to uncompress than bzip2
           # or an uncompressed tarball (and more bandwidth friendly).
-          safe_system "gzip", "-f", relocatable_tar_path
+          gz = Zlib::GzipWriter.open(bottle_path)
+          gz.mtime = tab.source_modified_time
+          gz.orig_name = relocatable_tar_path
+          gz.write File.binread(relocatable_tar_path)
```

Comment:
This is perhaps not RAM efficient - it reads the entire `tar` into memory and then writes it to gzip in one go.

I'll investigate chunking this read into smaller segments.